charts/kong/templates/service-kong-proxy.yaml

```diff
@@ -1,72 +1,53 @@
 metadata:
   name: {{ template "kong.fullname" . }}-proxy
   namespace: {{ template "kong.namespace" . }}
-  {{- if .Values.proxy.annotations }}
+  labels:
+    {{- include "kong.metaLabels" . | nindent 4 }}
+  {{- if .Values.proxy.ingress.annotations }}
   annotations:
-  {{- range $key, $value := .Values.proxy.annotations }}
+  {{- range $key, $value := .Values.proxy.ingress.annotations }}
     {{ $key }}: {{ $value | quote }}
   {{- end }}
   {{- end }}
-  labels:
-    enable-metrics: "true"
-    {{- include "kong.metaLabels" . | nindent 4 }}
 spec:
-  type: {{ .Values.proxy.type }}
-  {{- if eq .Values.proxy.type "LoadBalancer" }}
-  {{- if .Values.proxy.loadBalancerIP }}
-  loadBalancerIP: {{ .Values.proxy.loadBalancerIP }}
-  {{- end }}
-  {{- if .Values.proxy.loadBalancerSourceRanges }}
-  loadBalancerSourceRanges:
-  {{- range $cidr := .Values.proxy.loadBalancerSourceRanges }}
-  - {{ $cidr }}
-  {{- end }}
-  {{- end }}
-  {{- end }}
-  externalIPs:
-  {{- range $ip := .Values.proxy.externalIPs }}
-  - {{ $ip }}
-  {{- end }}
-  ports:
-  {{- if .Values.proxy.http.enabled }}
-  - name: kong-proxy
-    port: {{ .Values.proxy.http.servicePort }}
-    targetPort: {{ .Values.proxy.http.containerPort }}
-  {{- if (and (or (eq .Values.proxy.type "LoadBalancer") (eq .Values.proxy.type "NodePort")) (not (empty .Values.proxy.http.nodePort))) }}
-    nodePort: {{ .Values.proxy.http.nodePort }}
-  {{- end }}
-    protocol: TCP
-  {{- end }}
-  {{- if or .Values.proxy.tls.enabled }}
-  - name: kong-proxy-tls
-    port: {{ .Values.proxy.tls.servicePort }}
-    targetPort: {{ .Values.proxy.tls.overrideServiceTargetPort | default .Values.proxy.tls.containerPort }}
-  {{- if (and (or (eq .Values.proxy.type "LoadBalancer") (eq .Values.proxy.type "NodePort")) (not (empty .Values.proxy.tls.nodePort))) }}
-    nodePort: {{ .Values.proxy.tls.nodePort }}
-  {{- end }}
-    protocol: TCP
-  {{- end }}
-  {{- $root := .Values }}
-  {{- range .Values.proxy.stream }}
-  - name: stream-{{ .containerPort }}
-    port: {{ .servicePort }}
-    targetPort: {{ .containerPort }}
-    {{- if (and (or (eq $root.proxy.type "LoadBalancer") (eq $root.proxy.type "NodePort")) (not (empty .nodePort))) }}
-    nodePort: {{ .nodePort }}
-    {{- end }}
-    protocol: TCP
-  {{- end }}
-  {{- if .Values.proxy.externalTrafficPolicy }}
-  externalTrafficPolicy: {{ .Values.proxy.externalTrafficPolicy }}
-  {{- end }}
-  {{- if .Values.proxy.clusterIP }}
-  clusterIP: {{ .Values.proxy.clusterIP }}
-  {{- end }}
-  selector:
-    {{- include "kong.selectorLabels" . | nindent 4 }}
+  rules:
+    {{- range $host := .Values.proxy.ingress.hosts }}
+    - host: {{ $host | quote }}
+      http:
+        paths:
+          - path: {{ $path }}
+            backend:
+              serviceName: {{ $serviceName }}-proxy
+              servicePort: {{ $servicePort }}
+    {{- end -}}
+    {{- end -}}
+  {{- if .Values.proxy.ingress.tls }}
+  tls:
+{{ toYaml .Values.proxy.ingress.tls | indent 4 }}
```

Comment:
```suggestion
    {{- toYaml .Values.proxy.ingress.tls | indent 4 }}
```

Reply:
Indentation in the template does bad things to multi-line strings (it only affects the first line), and we do want to preserve the preceding newline here. Leaving as-is.